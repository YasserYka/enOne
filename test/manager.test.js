const manager = require("../lib/manager");

test("Load all plugins function", async () => {

    manager.load();

    manager.plugins.forEach(plugin => {
        
        expect(plugin).toMatchSnapshot({
            name: expect.any(String),
            config: expect.any(Object),
            object: expect.any(Object)
        });
    });
});

test("List local plugins", () => {

    const pluginNames = manager.listLocal();

    expect(pluginNames).not.toHaveLength(0);
});

test("List remote plugins", () => {

    const pluginNames = manager.listRemote();

    expect(pluginNames).not.toHaveLength(0);
});