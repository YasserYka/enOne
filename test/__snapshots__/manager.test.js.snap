// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load all plugins function 1`] = `
Object {
  "config": Any<Object>,
  "name": Any<String>,
  "object": Any<Object>,
}
`;
